contract/ELO.sol

```diff
@@ -0,0 +1,103 @@
+/**
+ * Simple ELO score rating system based on
+ * https://en.wikipedia.org/wiki/Elo_rating_system#Mathematical_details
+ */
+library ELO {
+    struct Score {
+        uint score;
+        string alias;
+    }
+    struct Scores {
+        mapping(address => Score) scores;
+    }
+
+    /**
+     * Records a game result for a game between two players.
+     * Winner can be 0 to record a draw
+     */
+    function recordResult(Scores storage self,
+                          address player1, address player2, address winner) {
+        // Get current scores
+        uint scoreA = getScore(self, player1);
+        uint scoreB = getScore(self, player2);
+
+        // Calculate result for player A
+        int resultA = 1; // 0 = lose, 1 = draw, 2 = win
+        if (winner == player1) {
+            resultA = 2;
+        }
+        else if (winner == player2) {
+            resultA = 0;
+        }
+
+        // Calculate new score
+        var (changeA, changeB) = getScoreChange(int(scoreA)-int(scoreB), resultA);
+        setScore(self, player1, uint(int(scoreA) + changeA));
+        setScore(self, player2, uint(int(scoreB) + changeB));
+    }
+
+    /**
+     * Table based expectation formula
+     * E = 1 / ( 1 + 10**((difference)/400))
+     * Table calculated based on inverse: difference = (400*log(1/E-1))/(log(10))
+     * scoreChange = K * (result - E)
+     * K = 20
+     * Because curve is mirrored around 0, uses only one table for positive side
+     * Returns (scoreChangeA, scoreChangeB)
+     */
+    function getScoreChange(int difference, int resultA) returns (int, int) {
+        bool reverse = (difference > 0); // note if difference was positive
+        uint diff = abs(difference); // take absolute to lookup in positive table
+        // Score change lookup table
+        int scoreChange = 10;
+        if (diff >= 798) scoreChange = 20;
```

Comment:
You are right, the number is kind of arbitrary. It is the score difference that results in a losing expectation of 1%, score change of 19.8. I thought it makes sense to have 20 as the maximum somewhere. Maybe it would be better to chose the point where score change is > 19.5?